### DISCUSSION TITLE
fixed buffer of any type

### DISCUSSION THREAD

**Discussion by ygc369** [dotnet/csharplang#6317] [Language Ideas]
This topic has been talked a lot. I just wonder whether it is still in .Net team's requirement list and when it will be implemented.

**Comment by huoyaoyuan** [comment]
This isn't something can be done solely at language side. For runtime it requires proper GC support for reference types. For language it requires accessing fixed-sized buffer via span instead of pointer.
Interestingly, these are both related to `stackalloc` for reference type.

**Comment by Xyncgas** [comment]
One frequent possible usage would be Encryption key being 32 bytes
But today you can not restrict a byte array to only hold 32 bytes in a class, you can do it in a struct with fixed statement

**Reply by theunrepentantgeek** [reply]
I don't think I've written code using encryption with a key as small as 256bits (32 bytes) in quite some time. 

Different ciphers have different key length requirements, so a fixed size buffer would be of limited use.

**Reply by qrli** [reply]
It is not hard to create a type which can be embedded in class:
```cs
public class C {
    private FixSizedKey encryptionKey1;
    ...
}

struct FixSizedKey {
    private int w1, w2, w3, w4;
    public unsafe Span<byte> AsSpan() => new Span<byte>(Unsafe.AsPointer(ref this), 32);
}
```

**Comment by Joe4evr** [comment]
Related: https://github.com/dotnet/runtime/issues/61135
